node/core/dispute-coordinator/src/scraping/mod.rs

```diff
@@ -92,24 +92,36 @@ impl Inclusions {
 		block_hash: Hash,
 	) {
 		if let Some(blocks_including) = self.inclusions_inner.get_mut(&candidate_hash) {
-			let first_lower_entry = blocks_including.partition_point(|(number_at_idx, _)| *number_at_idx >= block_number);
-			blocks_including.insert(first_lower_entry, (block_number, block_hash));
+			if let Some(blocks_at_height) = blocks_including.get_mut(&block_number) {
+				blocks_at_height.push(block_hash);
+			} else {
+				blocks_including.insert(block_number, Vec::from([block_hash]));
+			}
 		} else {
+			let mut blocks_including: BTreeMap<BlockNumber, Vec<Hash>> = BTreeMap::new();
+			blocks_including.insert(block_number, Vec::from([block_hash]));
 			self.inclusions_inner
-				.insert(candidate_hash, Vec::from([(block_number, block_hash)]));
+				.insert(candidate_hash, blocks_including);
 		}
 	}
 
 	pub fn remove_up_to_height(&mut self, height: &BlockNumber) {
-		for including_blocks in self.inclusions_inner.values_mut() {
-			let first_lower_entry = including_blocks.partition_point(|(number_at_idx, _)| *number_at_idx >= *height);
-			including_blocks.drain(first_lower_entry..);
+		for blocks_including in self.inclusions_inner.values_mut() {
+			*blocks_including = blocks_including.split_off(height);
 		}
-		self.inclusions_inner.retain(|_, including_blocks| including_blocks.len() > 0);
+		self.inclusions_inner.retain(|_, including_blocks| including_blocks.keys().len() > 0);
```

Comment:
The two different names (`blocks_including` and `including_blocks`) are throwing me off a bit. 🙃